plugins/GSdx/GSRendererHW.cpp

```diff
@@ -49,27 +49,29 @@ GSRendererHW::GSRendererHW(GSTextureCache* tc)
 
 void GSRendererHW::SetScaling()
 {
-	int buffer_size = max(m_context->FRAME.FBW * 64, m_regs->DISP[m_regs->PMODE.EN1 == 1 ? 0 : 1].DISPFB.FBW * 64);
-
-	//Only increase the buffer size, don't make it smaller, it breaks games (GH3)
-
-	// Also don't change the size for custom resolution (m_upscale_multiplier = 0).
-	if (m_upscale_multiplier && m_width < (buffer_size * m_upscale_multiplier)) {
-		m_tc->RemovePartial();
-	} else {
+	int crtc_width = GetDisplayRect().width(), crtc_height = GetDisplayRect().height();
+	int fb_width = max(m_context->FRAME.FBW * 64, m_regs->DISP[m_regs->PMODE.EN1 == 1 ? 0 : 1].DISPFB.FBW * 64);
+
+	// Framebuffer width is always a multiple of 64 so at certain cases it can't cover some weird width values.
+	// 480P , 576P use width as 720 which is not referencable by FBW * 64. so it produces 704 ( the closest value multiple by 64).
+	// In such cases, let's just use the CRTC width. the bigger the buffer , the better.
+	if (fb_width < crtc_width) fb_width = crtc_width;
```

Comment:
> ``the bigger the buffer , the better.``
No, we want the smallest buffest as possible to reduce memory consumption. (otherwise, fb = 1280 will be fine).

my bad. removed that stupid comment.